packages/Icon/src/createIconComponent.js

```diff
@@ -0,0 +1,59 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+export default glyphs => {
+  return class Icon extends PureComponent {
```

Comment:
Are we making this a class only for PureComponent usage as we're unsure if we can use React.memo?

We need to get clear minimum requirements for browsers and React. I'm still uncertain about these things, unless you know.

Reply:
It's a class for organizational reasons mostly. Static `displayName`, `propTypes`, and `defaultProps` are nice, and it also has a few getters that would be a bit messier in a functional component in my opinion. It's a class component less by necessity than because it's nicer to read.

Support for Memo is an issue, but given that these components will rarely re-render, this was not written as so for performance reasons.